### PR TITLE
Add ingress relation support and corresponding integration test

### PR DESCRIPTION
**Pull request by nrobinaubertin**
Adds the `ingress` interface support and the corresponding integration test.

We take as granted that the flask app will always be on port 8000.

### CODE REVIEW

**Comment by gregory-schiano** [comment]
Can you add a PR description ?

File: src/charm.py [hunk 1]
```diff
@@ -25,6 +26,12 @@ def __init__(self, *args: typing.Any) -> None:
         """
         super().__init__(*args)
         self.framework.observe(self.on.config_changed, self.config_service)
+        self.ingress = IngressPerAppRequirer(
+            self,
+            port=8000,
```
Comment (gregory-schiano) [inline src/charm.py]: I think we should define a constant for the port, it's repeated

File: src/charm.py [hunk 2]
```diff
@@ -25,6 +26,12 @@ def __init__(self, *args: typing.Any) -> None:
...
+            host=f"{self.app.name}-endpoints.{self.model.name}.svc.cluster.local",
```
Comment (gregory-schiano) [inline src/charm.py:37]: Can you add a comment for this that point to the bug that forces us to use this trick ?

File: pyproject.toml [hunk 3]
```diff
@@ -42,3 +42,9 @@ exclude_dirs = ["/venv/"]
 
 [tool.bandit.assert_used]
 skips = ["*/*test.py", "*/test_*.py"]
+
+[tool.mypy]
+ignore_missing_imports = true
```
Comment (arturo-seijas) [inline pyproject.toml:50]: I think this doesn't match what's defined in https://github.com/canonical/is-charms-contributing-guide/blob/main/CONTRIBUTING.md#type-hints

File: src/charm.py (hunk 2 above)
Comment (nrobinaubertin) [reply src/charm.py:37]: Added

File: src/charm.py (hunk 1 above)
Comment (nrobinaubertin) [reply src/charm.py]: Fixed

File: pyproject.toml (hunk 3 above)
Comment (nrobinaubertin) [reply pyproject.toml:50]: I sync'ed it with our guide but I still need to keep ignore_missing_imports for the ingress library

**Review by arturo-seijas — APPROVED** [review]

**Review by amandahla — APPROVED** [review]
LGTM

**Pull request merged**